File: Java/Digit/program19.java
//Accept two numbers from user and display the power

// Input : 2    4
// Output : 16  (2 * 2 * 2 * 2)

import java.util.*;

class Digit
{
    public int iNo1;
    public int iNo2;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the base");
        iNo1 = sobj.nextInt();

        System.out.println("Enter the power");
        iNo2 = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("The base is: "+this.iNo1);
        System.out.println("The power is: "+this.iNo2);
    }

    public int Power()
    {
        int iCnt = 0;
        int iMult = 1;

        for(iCnt = 1; iCnt <= iNo2; iCnt++)
        {
            iMult = iMult * iNo1;
        }
        return iMult;
    }
    
}

class program19
{
    public static void main(String arg[])
    {
        int iRet;

        Digit obj = new Digit();

        obj.Accept();
        obj.Display();

        iRet = obj.Power();
        System.out.println("Result is : "+iRet);
    }
}

File: C++/Digits/program19.cpp
//Accept two numbers from user and display the power

// Input : 2    4
// Output : 16  (2 * 2 * 2 * 2)

#include <iostream>
using namespace std;

class Number
{
    public:
        int iNo1;
        int iNo2;

        void Accept() // Setter
        {
            cout << "Enter the base" << endl;
            cin >> iNo1;

            cout << "Enter the power" << endl;
            cin >> iNo2;
        }

        void Display() // Getter
        {
            cout << "The base is: " << this->iNo1 << endl;
            cout << "The power is: " << this->iNo2 << endl;
        }

        int Power()
        {
            int iCnt = 0;
            int iMult = 1;

            for(iCnt = 1; iCnt <= iNo2; iCnt++)
            {
                iMult = iMult * iNo1;
            }
            return iMult;
        }
};

int main()
{
    int iRet;

    Number nobj;

    nobj.Accept();
    nobj.Display();

    iRet = nobj.Power();
    cout<<"Result is : "<<iRet<<endl;
    
    return 0;
}